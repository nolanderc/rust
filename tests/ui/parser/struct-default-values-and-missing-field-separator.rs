//@ run-rustfix
#![allow(dead_code)]

enum E {
    A,
}

struct S {
    field1: i32 = 42,
    field2: E = E::A,
    field3: i32 = 1 + 2,
    field4: i32 = { 1 + 2 },
    field5: E = foo(42),
    field6: E = { foo(42) },
}

struct S1 {
    field1: i32 //~ ERROR expected `,`, or `}`, found `field2`
    field2: E //~ ERROR expected `,`, or `}`, found `field3`
    field3: i32 = 1 + 2,
    field4: i32 = { 1 + 2 },
    field5: E = foo(42),
    field6: E = { foo(42) },
}

struct S2 {
    field1 = i32, //~ ERROR expected `:`, found `=`
    field2; E, //~ ERROR expected `:`, found `;`
}

const fn foo(_: i32) -> E {
    E::A
}

fn main() {}
