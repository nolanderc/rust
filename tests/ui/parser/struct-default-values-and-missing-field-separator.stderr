error: expected `,`, or `}`, found `field2`
  --> $DIR/struct-default-values-and-missing-field-separator.rs:18:16
   |
LL |     field1: i32
   |                ^ help: try adding a comma: `,`

error: expected `,`, or `}`, found `field3`
  --> $DIR/struct-default-values-and-missing-field-separator.rs:19:14
   |
LL |     field2: E
   |              ^ help: try adding a comma: `,`

error: expected `:`, found `=`
  --> $DIR/struct-default-values-and-missing-field-separator.rs:27:12
   |
LL |     field1 = i32,
   |            ^
   |            |
   |            expected `:`
   |            help: field names and their types are separated with `:`

error: expected `:`, found `;`
  --> $DIR/struct-default-values-and-missing-field-separator.rs:28:11
   |
LL |     field2; E,
   |           ^
   |           |
   |           expected `:`
   |           help: field names and their types are separated with `:`

error: aborting due to 4 previous errors

